### ISSUE TITLE
Add missing `align(With)` functions

### ISSUE THREAD

**Issue by gallais** [agda/agda-stdlib#383]
See `align(With)` defined in `Codata.Delay` in #382 

```agda
alignWith : (These A B → C) → ∀ {i} → Delay A i → Delay B i → Delay C i
align : ∀ {i} → Delay A i → Delay B i → Delay (These A B) i
```

We have opportunities for these in at least:

* [x] Data.Maybe
* [x] Data.List
* [x] Data.Vec
* [x] Data.List.NonEmpty
* [x] Data.BoundedVec
* [x] Codata.Colist

**Comment by gallais** [comment]
Blocked by #397

**Comment by MatthewDaggitt** [comment]
Removing this from release as blocked.

**Comment by MatthewDaggitt** [comment]
Closed with 1dde7ff
